iceoryx_posh/source/runtime/service_discovery.cpp

```diff
@@ -89,6 +89,19 @@ ServiceDiscovery::findService(const cxx::optional<capro::IdString_t>& service,
     return {cxx::success<ServiceContainer>(serviceContainer)};
 }
 
+void ServiceDiscovery::findService(const cxx::optional<capro::IdString_t>& service,
+                                   const cxx::optional<capro::IdString_t>& instance,
+                                   const cxx::optional<capro::IdString_t>& event,
+                                   const cxx::function_ref<void(const ServiceContainer&)>& callable) noexcept
+{
+    /// @todo #415 change implementation once PR #1088 is merged
+    auto searchResult = findService(service, instance, event);
+    if (!searchResult.has_error())
+    {
+        callable(searchResult.value());
```

Comment:
What if `callable` is not set?

Please add this to the beginning of the function:
```
if ( !callable ) {
  return;
}
```